### PR TITLE
surface more exception info for call creds errors

### PR DESCRIPTION
**Pull request by apolcyn**
addresses https://github.com/grpc/grpc/issues/8072

### CODE REVIEW

**Comment by apolcyn** [comment]
the ruby failure on opt/linux looks like it's running into the bidi test issue https://github.com/grpc/grpc/issues/7972

File: src/ruby/ext/grpc/rb_call_credentials.c [hunk 1]
```diff
@@ -86,19 +86,16 @@ static VALUE grpc_rb_call_credentials_callback_rescue(VALUE args,
...
-          exception_classname, StringValueCStr(rb_exception_info),
+  gpr_log(GPR_INFO, "Call credentials callback failed: %s\n%s",
+          StringValueCStr(rb_exception_info),
           StringValueCStr(backtrace));
   rb_hash_aset(result, rb_str_new2("metadata"), Qnil);
-  /* Currently only gives the exception class name. It should be possible get
-     more details */
   rb_hash_aset(result, rb_str_new2("status"),
-               INT2NUM(GRPC_STATUS_PERMISSION_DENIED));
+               INT2NUM(GRPC_STATUS_UNAVAILABLE));
```
Comment (murgatroid99) [inline src/ruby/ext/grpc/rb_call_credentials.c]: It looks like this should be `GRPC_STATUS_UNAUTHENTICATED` (based on the test).

Comment (apolcyn) [inline src/ruby/ext/grpc/rb_call_credentials.c]: It looks like `on_credentials_metadata`, which I believe is the actual function for the `grpc_credentials_plugin_metadata_cb` callback, in https://github.com/grpc/grpc/blob/master/src/core/lib/security/transport/client_auth_filter,always bubbles up UNAUTHENTICATED as long as status passed to it isn't OK.

Was thinking the status passed to be the result status for the call, but is thie behavior the expectation of the credentials plugin and callback? cc @ctiller

**Comment by apolcyn** [comment]
test this please

**Comment by apolcyn** [comment]
test this please

**Comment by apolcyn** [comment]
test this please

**Comment by apolcyn** [comment]
test this please

**Pull request merged**